# Graded assignment 1

Mark Klik & Misja Mikkers


# Packages

We need the package `tidyverse`

In [43]:
library(tidyverse) 

# Assignment 1

First, we generate some random data:

In [44]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1        V2         V3         V4        V5        V6         V7       
1 0.6524039 0.38769374 0.07740113 0.3765009 0.7866352 0.74614551 0.2455575
2 0.1054899 0.03531505 0.48561027 0.7144598 0.6902666 0.09778803 0.6140648
  V8        V9        V10       V11       V12       V13       V14       
1 0.3198175 0.2943395 0.0210569 0.8831167 0.1375629 0.8944962 0.91745278
2 0.6066412 0.8071123 0.5134313 0.7412320 0.9578095 0.4165038 0.06289122
  V15       V16       V17       V18       V19       V20       
1 0.9336507 0.2856958 0.0496773 0.3523508 0.6202251 0.95389955
2 0.2221775 0.2032293 0.8296626 0.6179436 0.8912231 0.01915682

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [45]:
df2 <- df1 %>% gather(Key, Value, V1:V20)
  

head(df2)

Key Value     
1 V1  0.65240392
2 V1  0.10548992
3 V2  0.38769374
4 V2  0.03531505
5 V3  0.07740113
6 V3  0.48561027

## Assignment 2

We will generate some data:

In [46]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [47]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [48]:
df5 <- fulljoin <- full_join(df3, df4, by = 'id')
df5

id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook